test/micro/org/openjdk/bench/vm/compiler/VectorStoreToLoadForwarding.java

```diff
@@ -0,0 +1,142 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.bench.vm.compiler;
+
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.*;
+
+import java.lang.invoke.*;
+
+import java.util.concurrent.TimeUnit;
+import java.util.Random;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Thread)
+@Warmup(iterations = 2, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 3, time = 1, timeUnit = TimeUnit.SECONDS)
+@Fork(value = 1)
+public abstract class VectorStoreToLoadForwarding {
+    @Param({"10000"})
+    public int SIZE;
+
+    @Param({  "0",   "1",   "2",   "3",   "4",   "5",   "6",   "7",   "8",   "9",
+             "10",  "11",  "12",  "13",  "14",  "15",  "16",  "17",  "18",  "19",
+             "20",  "21",  "22",  "23",  "24",  "25",  "26",  "27",  "28",  "29",
+             "30",  "31",  "32",  "33",  "34",  "35",  "36",  "37",  "38",  "39",
+             "40",  "41",  "42",  "43",  "44",  "45",  "46",  "47",  "48",  "49",
+             "50",  "51",  "52",  "53",  "54",  "55",  "56",  "57",  "58",  "59",
+             "60",  "61",  "62",  "63",  "64",  "65",  "66",  "67",  "68",  "69",
+             "70",  "71",  "72",  "73",  "74",  "75",  "76",  "77",  "78",  "79",
+             "80",  "81",  "82",  "83",  "84",  "85",  "86",  "87",  "88",  "89",
+             "90",  "91",  "92",  "93",  "94",  "95",  "96",  "97",  "98",  "99",
+            "100", "101", "102", "103", "104", "105", "106", "107", "108", "109",
+            "110", "111", "112", "113", "114", "115", "116", "117", "118", "119",
+            "120", "121", "122", "123", "124", "125", "126", "127", "128", "129"})
+    public int OFFSET;
+
+    // To get compile-time constants for OFFSET
+    static final MutableCallSite MUTABLE_CONSTANT = new MutableCallSite(MethodType.methodType(int.class));
+    static final MethodHandle MUTABLE_CONSTANT_HANDLE = MUTABLE_CONSTANT.dynamicInvoker();
+
+    public int START = 1000;
+
+    private byte[] aB;
+    private short[] aS;
+    private int[] aI;
+    private long[] aL;
+
+    @Param("0")
+    private int seed;
+    private Random r = new Random(seed);
+
+    @Setup
+    public void init() throws Throwable {
+        aB = new byte[SIZE];
+        aS = new short[SIZE];
+        aI = new int[SIZE];
+        aL = new long[SIZE];
+
+        for (int i = START; i < SIZE; i++) {
+            aB[i] = (byte)r.nextInt();
+            aS[i] = (short)r.nextInt();
+            aI[i] = r.nextInt();
+            aL[i] = r.nextLong();
+        }
+
+        MethodHandle constant = MethodHandles.constant(int.class, OFFSET);
+        MUTABLE_CONSTANT.setTarget(constant);
+    }
+
+    @CompilerControl(CompilerControl.Mode.DONT_INLINE)
```

Comment:
Err are you sure this works I think this should be `FORCE_INLINE` instead. I see you want to have different `SIZE`s, too. Then you can make a `MutableCallSite` for each parameter. The magic here is that the compiler will treat the call target as a constant and force a recompilation each time you call `setTarget`

Reply:
I don't need different sizes though. Just the OFFSET is ok - those really need to be constant.